src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1971,10 +1971,10 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
     }
 
     private def ensurePredefParentsAreInSameSourceFile(template: Template) = {
-      val parentSyms = template.parents map (_.symbol) filterNot (_ == AnyRefClass)
+      val parentSyms = template.parents.collect { case parent if parent.symbol != AnyClass => parent.symbol }
```

Comment:
We can merge this `collect` with the `exists` below.
```suggestion
      val hasUndefinedParent = template.parents.exists { parent => 
        parent.symbol != AnyClass  && parent.symbol.associatedFile != PrefedModuleFile
      }
```